tsfile/src/main/java/org/apache/iotdb/tsfile/utils/PublicBAOS.java

```diff
@@ -61,7 +61,7 @@ public PublicBAOS(int size) {
    * @exception  IOException  if an I/O error occurs.
    */
   @Override
-  public void writeTo(OutputStream out) throws IOException {
+  public synchronized void writeTo(OutputStream out) throws IOException {
```

Comment:
Hi, this is removed on purpose, cause this will impact the Flush pipeline.

See: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=173085039

Reply:
Got it, I prefer to post this link here in this function, so that other developers know about the problem, otherwise, sonar will report a bug, and other developers may fix it.